src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LanguageServerProjectSystem.cs

```diff
@@ -205,6 +206,15 @@ private async Task<(LSP.MessageType? failureType, BuildHostProcessKind? preferre
             if (await buildHost.IsProjectFileSupportedAsync(projectPath, cancellationToken))
             {
                 var loadedFile = await buildHost.LoadProjectFileAsync(projectPath, cancellationToken);
+                var diagnosticLogItems = await loadedFile.GetDiagnosticLogItemsAsync(cancellationToken);
+                LogDiagnostics(projectPath, diagnosticLogItems);
+
+                if (diagnosticLogItems.Any(item => item.Kind is WorkspaceDiagnosticKind.Failure))
```

Comment:
I don't know if this is the right thing to do - but seems reasonable.  In the issue linked in the description if we don't do this, we'll get the initial error, and then it throws again afterwards in `GetProjectFileInfosAsync`

Reply:
So this will catch the evaluation errors, but execution level errors will get dropped here since those won't be found until the later call to GetProjectFileInfosAsync(). So the early bail is probably good, but you need to leave the latter checks in too.

Reply:
Oh - the collection can get updated underneath?  I see that does make it more complicated.